app/models/data_resources/event_record.rb

```diff
@@ -199,6 +226,10 @@ def list_date
     calculated_list_date
   end
 
+  def upcoming_dates
+    dates.upcoming
+  end
+
```

Comment:
maybe cleaner with `delegate`

```
class EventRecord < ApplicationRecord
  has_many :dates

  delegate :upcoming, to: :dates, prefix: true
end
```

=> 

event_record.dates_upcoming (instead of event_record.upcoming_dates in EventRecordType)

Reply:
the way is unknown to me. can you create a commit for it?

Reply:
https://github.com/smart-village-solutions/smart-village-app-mainserver/pull/431/commits/e9185f3e17f8c49c0117676ddf04af71915b6f09